### PR TITLE
[nested] add check before launching L2 vm

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/tools/qemu.py [hunk 1]
```diff
@@ -55,12 +57,14 @@ def create_vm(
 
         # store name of tap interfaces added to the vm
         added_taps: List[str] = []
-
+        cmd = ""
+        if timeout > 0:
+            cmd = f"timeout {timeout}  {self.command} "
```
Comment (dsrivastavv) [inline lisa/tools/qemu.py:62]: Can we use timeout in `self.run` directly?

File: lisa/tools/qemu.py [hunk 2]
```diff
@@ -95,21 +99,35 @@ def create_vm(
...
-        # start qemu
-        self.run(
-            cmd,
-            sudo=True,
-            shell=True,
-            expected_exit_code=0,
-            expected_exit_code_failure_message=f"Unable to start VM {guest_image_path}",
-        )
+        if timeout > 0:
+            cmd_result = self.node.execute(
```
Comment (dsrivastavv) [inline lisa/tools/qemu.py:112]: Does this command exit after starting up qemu if `demonize` is not present?

File: microsoft/testsuites/nested/common.py [hunk 3]
```diff
@@ -67,6 +68,21 @@ def qemu_connect_nested_vm(
...
+            nic_model=nic_model,
+            taps=taps,
+            bridge=bridge,
+            disks=disks,
+            cores=cores,
+            stop_existing_vm=stop_existing_vm,
+            run_in_background=False,
+            timeout=60,
+        )
+        qemu.interface_count = 0
```
Comment (dsrivastavv) [inline microsoft/testsuites/nested/common.py:85]: Why is this needed?

**Comment by LiliDeng** [comment]
after discussing with @somil55 , he will take care of this.
